digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/TierekisteriClient.scala

```diff
@@ -160,14 +160,38 @@ case class TierekisteriMassTransitStop(nationalId: Long,
 
 trait TierekisteriAssetData {
   val roadNumber: Long
-  val roadPartNumber: Long
-  val starMValue: Double
-  val endMValue: Double
+  val startRoadPartNumber: Long
+  val endRoadPartNumber: Long
+  val startAddressMValue: Long
+  val endAddressMValue: Long
 }
 
-case class TierekisteriTrafficData(roadNumber: Long, roadPartNumber: Long, starMValue: Double, endMValue: Double, kvl: Int) extends TierekisteriAssetData
-
-case class TierekisteriLighting(roadNumber: Long, roadPartNumber: Long, starMValue: Double, endMValue: Double) extends TierekisteriAssetData
+case class AddressSection(roadNumber: Long, roadPartNumber: Long, track: Track, startAddressMValue: Long, endAddressMValue: Option[Long])
+
+case class TierekisteriTrafficData(roadNumber: Long, startRoadPartNumber: Long, endRoadPartNumber: Long,
+                                   startAddressMValue: Long, endAddressMValue: Long, kvl: Int) extends TierekisteriAssetData
+
+case class TierekisteriLighting(roadNumber: Long, startRoadPartNumber: Long, endRoadPartNumber: Long,
+                                track: Track, startAddressMValue: Long, endAddressMValue: Long) extends TierekisteriAssetData {
+  val getRoadAddressSections: Seq[AddressSection] = {
+    Seq(AddressSection(roadNumber, startRoadPartNumber, track, startAddressMValue,
+      if (endRoadPartNumber == startRoadPartNumber)
+        Some(endAddressMValue)
+      else
+        None)) ++ {
+      if (startRoadPartNumber != endRoadPartNumber)
+        (startRoadPartNumber until endRoadPartNumber).tail.map(part => AddressSection(roadNumber, part, track, 0L, None))
+      else
+        Seq[AddressSection]()
+    } ++
+      {
+        if (endRoadPartNumber == startRoadPartNumber)
+          Seq[AddressSection]()
+        else
+          Seq(AddressSection(roadNumber, endRoadPartNumber, track, 0L, Some(endAddressMValue)))
+      }
+  }

```

Comment:
This can be improved because we are always checking the same (endRoadPartNumber == startRoadPartNumber)